docs/modeling-your-data/modeling-your-data-with-dbt/index.md

```diff
@@ -105,3 +105,112 @@ From version v0.13.0 onwards you must be using [RDB Loader](/docs/destinations/w
+
+```mdx-code-block
+import { dbtVersions } from '@site/src/dbtVersions';
+import { satisfies, gt, valid } from "semver";
+import { useState } from 'react';
+import TextField from '@mui/material/TextField';
+
+
+
+
+export function VersionChecker() {
+    const [activeIndex, setActiveIndex] = useState(0);
+    const [dbtVer, setDbtVer] = useState('')
+    const [utilsVer, setutilsVer] = useState('')
+    return (
+        <>
+        <TextField
+            id="input-dbt-version"
+            label="dbt Version"
+            onChange={() => setDbtVer(event.target.value)}
+            />
+            &nbsp; &nbsp;
+            <TextField
+            id="input-utils-version"
+            label="dbt-utils Version"
+            onChange={() => setutilsVer(event.target.value)}
+            />
+        <br/>
+        <br/>
+        <Getsupportedpackages dbtVer={dbtVer} utilsVer={utilsVer}/>
+        </>
+    );
+}
+
+
+export const Getsupportedpackages = ({children, dbtVer, utilsVer}) => {
+    let packageVersions = []
+    if (! valid(dbtVer)) {
+        return(<>
+        <i>Please enter a valid dbt version.</i>
+        </>)
+    }
+    if (! valid(utilsVer) && utilsVer != '') {
+        return(<>
+        <i>Please enter a valid dbt-utils version.</i>
+        </>)
+    }
+    if (utilsVer === '') {
+        for(const [pkg, version] of Object.entries(dbtVersions)) {
+            let maxVer = '0.0.0'
+            let actualMaxVer = '0.0.0'
+            for(const [ver, details] of Object.entries(version)) {
+                {/* Check if the version is in the range the package supports, and check if this version is newer than our previous highest */}
+                if (satisfies(dbtVer, details['dbtversion'], {'includePrerelease':true}) && gt(ver, maxVer) ) {
+                    maxVer = ver
+                }
+                if (gt(ver, actualMaxVer)) {
+                    actualMaxVer = ver
+                }
+            }
+            if (maxVer == '0.0.0') {
+                packageVersions.push({'package': pkg, 'maxVer': 'none', 'actualMaxVer' : actualMaxVer})
+            } else {
+                packageVersions.push({'package': pkg, 'maxVer': maxVer, 'actualMaxVer' : actualMaxVer})
+            }
+        }
+        return (
+        <>
+        <p>For <b>dbt</b> version <code>{dbtVer}</code> the latest version of each of our packages you can install are:</p>
+        {packageVersions.map(x => <li>{x['package']}: <code>{x['maxVer'] == 'none' ? 'No matching version supported' : x['maxVer'] }</code> &nbsp; &nbsp; <i>(latest version: <code>{x['actualMaxVer']}</code>)</i></li>)}
+        </>
+        )
+    } else {
+        for(const [pkg, version] of Object.entries(dbtVersions)) {
```

Comment:
1. Same comments as above
2. This seems to be 99% the same, can we avoid duplication?

Reply:
I got rid of most of it, I can't work out how to conditionally add the string in the first line of the return so there's still some duplication there...